src/wallet/db.cpp

```diff
@@ -826,6 +826,10 @@ void BerkeleyDatabase::Flush(bool shutdown)
             LOCK(cs_db);
             g_dbenvs.erase(env->Directory().string());
             env = nullptr;
+        } else {
+            // To avoid accessing a map that has already deconstructed, do not call this when shutdown is true. g_dbenvs.erase would also erase this value.
+            // TODO: get rid of wild pointers
+            env->m_fileids.erase(strFile);
```

Comment:
It might not be clear what "wild pointers" refers to in this todo (laanwj asked about it here: https://github.com/bitcoin/bitcoin/pull/14531#discussion_r227770685). Wild pointers could mean either dead pointers or raw pointers, but I think the TODO is referring to the bug described here: https://github.com/bitcoin/bitcoin/pull/14320#discussion_r220332273. Maybe could be expanded as:

```c++
// TODO: To avoid g_dbenvs.erase erasing the environment prematurely after the
// first database shutdown when multiple databases are open in the same
// environment, should replace raw database `env` pointers with shared or weak
// pointers, or else separate the database and environment shutdowns so
// environments can be shut down after databases.
```